platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbStatisticsUpdater.java

```diff
@@ -48,22 +42,18 @@ public class MerkleDbStatisticsUpdater {
             .withFormat("%d");
 
     private final MerkleDbStatistics statistics;
-    private final MerkleDbDataSource<?, ?> dataSource;
```

Comment:
This is the core of the fix. Since `MerkleDbStatisticsUpdater` may be reused across different `MerkleDbDataSource` instances, they can't hold a reference to a data source, otherwise the described memory leak would be observed. Instead, a data source instance is passed to all updater methods, so it knows where to pull stats from.